### PR TITLE
[GLT-3660] added SAML authentication

### PR DESCRIPTION
**Pull request by djcooke**

### CODE REVIEW

File: example-application.properties [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# This file should be copied and filled in to provide instance-specific configuration
+
```
Comment (mlaszloffy) [inline example-application.properties:2]: server.port=8081

File: src/main/java/ca/on/oicr/gsi/dimsum/SecurityConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+@Profile("!noauth")
+public class SecurityConfiguration {
+
+  private static final String LOGIN_URL = "/login";
+
+  @Bean
+  public SecurityFilterChain securityFilterChain(HttpSecurity http) throws Exception {
+    http
+        .authorizeHttpRequests(auth -> auth
+            .antMatchers("/favicon.ico").permitAll()
```
Comment (mlaszloffy) [inline src/main/java/ca/on/oicr/gsi/dimsum/SecurityConfiguration.java:36]: /metrics

**Review by mlaszloffy — APPROVED** [review]

File: example-application.properties (hunk 1 above)
Comment (djcooke) [reply example-application.properties:2]: Not required. If you're not already running Tomcat (or anything else on port 8080) then the default is fine. For production, we'll most likely put NGINX in front of it, so the default port is fine.

File: src/main/java/ca/on/oicr/gsi/dimsum/SecurityConfiguration.java (hunk 2 above)
Comment (djcooke) [reply src/main/java/ca/on/oicr/gsi/dimsum/SecurityConfiguration.java:36]: added

**Review by asofmosq — APPROVED** [review]

**Pull request merged**